Dockerfile

```
@@ -0,0 +1,49 @@
+#
+# build by: docker build -t sniproxy .
+#
+
+FROM ubuntu:20.04 AS builder
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV TZ Asia/Tehran
+
+# Build Stage
+RUN apt-get update && \
+    apt-get install -y \
+        autotools-dev \
+        cdbs \
+        debhelper \
+        dh-autoreconf \
+        dpkg-dev \
+        gettext \
+        libev-dev \
+        libpcre3-dev \
+        libudns-dev \
+        pkg-config \
+        fakeroot \
+        devscripts && \
+    mkdir -p /sniproxy
+
+COPY . /sniproxy/
+
+RUN cd /sniproxy && \
+    ./autogen.sh && \
+    dpkg-buildpackage
+
+
+
+# Usage
+FROM ubuntu:20.04
+
+COPY --from=builder /*.deb /tmp/
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        libev-dev \
+        libudns-dev \
+        libpcre3-dev && \
+    dpkg -i /tmp/sniproxy_*.deb && \
```

I'd suggest to install the `.deb` with `apt` like

```suggestion
    apt install /tmp/sniproxy_*.deb && \
```

Because it will then install the dependencies for the package as declared by the deb-file itself. https://unix.stackexchange.com/a/159114/23363